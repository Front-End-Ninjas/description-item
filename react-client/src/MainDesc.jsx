import React from 'react';
import PropTypes from 'prop-types';
import ListDesc from './ListDesc';

const MainDesc = ({ dataDesc }) => (
  <div id="main_desc">
    <div>
      <p id="brand">{dataDesc.brand}</p>
    </div>
    <div>
      <h1 id="model">{dataDesc.model}</h1>
    </div>
    <div>
      <p id="price_shipping">Price: ${dataDesc.price} & {dataDesc.shipping === 'Prime' ?
        <img src="../prime_logo.png" alt="Amazon Prime" /> : dataDesc.shipping }
      </p>
    </div>
    <div>
      <h3 id="stock">{dataDesc.stock ? 'In Stock' : 'Temporarily Out of Stock'}</h3>
    </div>
    <div id="list_desc">
      {dataDesc.description.map(x => <ListDesc item={x} key={x} />)}
    </div>
  </div>
);

MainDesc.propTypes = {
  dataDesc: PropTypes.shape({
    brand: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    shipping: PropTypes.string.isRequired,
    stock: PropTypes.bool.isRequired,
    description: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }).isRequired,
};

export default MainDesc;
