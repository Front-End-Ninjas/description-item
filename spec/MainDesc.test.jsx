import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MainDesc from '../react-client/src/MainDesc';

describe('MainDesc Component', () => {
  const data = {
    brand: 'brandTest',
    model: 'modelTest',
    price: 100,
    shipping: 'shippingTest',
    stock: true,
  };

  it('should be defined', () => {
    expect(MainDesc).toBeDefined();
  });

  it('should render a Brand <p> tag', () => {
    expect(shallow(<MainDesc dataDesc={data} />).find('#brand').length).toEqual(1);
  });

  it('should render a Model <h1> tag', () => {
    expect(shallow(<MainDesc dataDesc={data} />).find('#model').length).toEqual(1);
  });

  it('should render a Price and Shipping <p> tag', () => {
    expect(shallow(<MainDesc dataDesc={data} />).find('#price_shipping').length).toEqual(1);
  });

  it('should render a Stock <h3> tag', () => {
    expect(shallow(<MainDesc dataDesc={data} />).find('#stock').length).toEqual(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MainDesc dataDesc={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
