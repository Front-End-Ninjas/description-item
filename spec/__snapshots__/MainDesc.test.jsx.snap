// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainDesc Component should renders correctly 1`] = `
<div
  className="MainDesc"
>
  <div>
    <p
      id="brand"
    >
      brandTest
    </p>
  </div>
  <div>
    <h1
      id="model"
    >
      modelTest
    </h1>
  </div>
  <div>
    <p
      id="price_shipping"
    >
      Price: $
      100
       & 
      shippingTest
    </p>
  </div>
  <div>
    <h3
      id="stock"
    >
      In Stock
    </h3>
  </div>
  <div>
    <div>
      <ul>
        <li
          id="list-item"
        >
          testItem
        </li>
      </ul>
    </div>
  </div>
</div>
`;
